### PR TITLE
New site with gatsby

### LINKED ISSUE #367: Migrate website to gatsby and redesign.

### PR DESCRIPTION
**Pull request by mitchellhamilton**
**What**:

A new site.

**Why**:

Closes #367

**How**:

gatsby.

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Code complete

This isn't close to being finished but you can check it out at https://gatsby.emotion.sh

### CODE REVIEW

**Comment by tkh44** [comment]
**Checklist**:
[ ] - set up our theme. 
 - We should have a theme.js with what is required for styled-system and our colors.

[ ] - design. 
- @mitchellhamilton and I are not the best designers 😅. We need help. Please comment with any ideas or inspirations you might want to share. 

[ ] - code examples

[ ] - documentation organization

[ ] - documentation updates for version 8
- This includes proof reading as well as contributing
- If you hit a hurdle in the upgrade and found a way around it please open a PR and share with everyone.

**Comment by souporserious** [comment]
Do you have an outline for the design? I may be able to squeeze something out this week 😁

**Comment by tkh44** [comment]
I think the first thing we need to figure out is some colors that work with the logo image.



I'm not sure how to coordinate a set of colors that work with the logo and not against it.

@mitchellhamilton has done a great job with the general structure. We just need to expand on what he's started and add those small details.

For those interested in emotion this is a pretty fun site to tinker with because it uses a component called `Box`. Based on [styled-system](https://github.com/jxnblk/styled-system), this component will let you write responsive styles with minimal effort. (Just try it 😄 )

---

Here are some doc sites I personally like and could be useful to draw inspiration from.

https://reasonml.github.io/
- general layout
- examples are shown clearly with screenshots
- the REPL

https://reactjs.org/
- code examples down the page 
- search is great

https://svelte.technology
- the REPL

**Comment by norbertdragan** [comment]
Hey everyone 👋 ,

I'm new here. Saw @tkh44's [tweet](https://twitter.com/tkh44/status/914699408570245120) about how you're looking for some design contribution and I though I would jump onboard. Messaged @tkh44, and he said to introduce myself here.

I'm 29, from Hungary, currently living in Bali. I work as a freelance web designer/developer, my site is at [dragan.io](http://dragan.io).

I did not know about emotion before, and I have never contributed to an open source project before. However, I use a bunch of them, including gatsby, which I'm a big fan of.

I'll study a bit emotion and try it out and see what I can do. As an open source noob, any kind of guidence is appreciated. 😃 But all in all, I have some time, and I want to grow my portfolio, so this could be a good match.

**Comment by norbertdragan** [comment]
Hey,

I did a a super quick color exploration with the logo:

emotion_color_exploration

I think this is a nice color palette and the gradient variatons could also work well. Personally, I like working with gradients. Styled-components uses something similar to gradient nr. 3, so I wouldn't use that. Perhaps the first is the best.

**Comment by tkh44** [comment]
That's awesome @norbertdragan. I think I may know a good way to use the gradient without overdoing it.

**Comment by norbertdragan** [comment]
So I went ahead and designed a couple pages. The landing is a bit edgy with the dark background, but I think it's okay considering that the topic is CSS. If there were more content to add to the landing, I would use a light background for them, though.

I was assuming there are no logo guidelines, sorry if some butchering occurred. 😬 For the font, I stick to Oxygen. The syntax highlighting color is just pasted from my editor, that is something I would change for sure.

Is this a direction you would consider for the project? Since I didn't do much research, I totally understand if not. But it was fun making these two pages nonetheless.

Let me know what you think! 

emotion_landing_v1
emotion_docs_v1

**Comment by tkh44** [comment]
Wow!! 

These are looking great @norbertdragan. I really like the direction you took and the use of the darker background.

**Comment by tkh44** [comment]
I wonder if we could make the background of the text use the gradient like https://dribbble.com/shots/3834566-Section-32-Homepage. Not sure if it would work on the dark background though.

**Comment by evan-scott-zocdoc** [comment]
@norbertdragan wow that looks amazing

File: packages/site/src/docs/configurable-imports.md [hunk 1]
```diff
@@ -43,7 +45,7 @@ One use case for this functionality is to migrate incrementally from a
 styled-jsx application. When compiling the following file with emotion
 and styled-jsx.
 
-```js
+```jsx
```
Comment (tkh44) [inline packages/site/src/docs/configurable-imports.md]: we should make the `javascript`. It will code highlight in more places.

Comment (mitchellhamilton) [reply packages/site/src/docs/configurable-imports.md]: By default Prism only highlights JSX with the jsx language so we would have to change some stuff so IMO we should leave it as jsx.

**Comment by norbertdragan** [comment]
I think the landing page could benefit from a few new sections that explain the benefits of emotion.

But to do better desing work, I would like to do some research. It would be great if I could speak to some of your users to understand _why_ are they using emotion over other solutions. 

Or is there thread somewhere where to community answered this already? Or any resource online that I should check out?

How would you describe the reasons for people using emotion over other solutions?

**Comment by norbertdragan** [comment]
For the record, here's the corresponding tweet: https://twitter.com/NorbertDragan/status/915752775400570880

**Comment by tkh44** [comment]
I had a bit of time tonight so I went and did some rough work on the general layout. 
emotion-site

(I also may have had a little fun with animations as well)

**Comment by tgrecojs** [comment]
@tkh44 I just spent a little bit looking at the code base and I think I can definitely be of assistance out here 😄. I'm still a novice open-source contributor so I wanted to reach out before I made a PR for any code however I did spot a minor error on the home page regarding the `active` props each button is getting on click.   - [pages/index.js](https://github.com/emotion-js/emotion/blob/1e992d69f56da5b55bd53db005c3e639dfe5709a/packages/site/src/pages/index.js#L89). 

I can definitely help provide some assistance in wiring up @norbertdragan's (awesome 😎) design but I am just wondering the best way to go about it. Specifically, I see that you already have a layout using the css grid some-what in place so i'm wondering if you're looking for that to be the base or if you're open to checking out other layouts. 

I know that this is in the early stages but I'm also a super big advocate for writing modular JS apps using factory functions as opposed to classes, so I would be happy to provide some help refactoring code, writing tests, etc. Again, people program all different sorts of ways so before I really dove in and begin coding I thought it would be wise to reach out. Excited to hear back / contribute to such an awesome project 😄  💪

**Comment by norbertdragan** [comment]
I have made a few changes to the typography, layout and buttons. First PR, I hope I'm doing it right!

https://github.com/emotion-js/emotion/pull/382

**Comment by tkh44** [comment]
@tgrecojs and anyone else interested, if you see a bug and want to fix it don't be afraid to submit a PR. The absolute worst that will happen is... we don't accept the PR. It happens, but you've now become a familiar face that we know is serious about helping 👍 The best way is it make **small** prs against this branch. I would suggest staying within the style of code that we've already started. Lets focus on those small little details that add up big in the end.

**Comment by tkh44** [comment]
Could we add "Edit this page" buttons or links to the top of the docs pages that link to their edit location on github?

Like this: https://github.com/emotion-js/emotion/edit/gatsby/packages/site/src/docs/babel-macros.md

**Comment by KyleAMathews** [comment]
Saw [Kye's tweet](https://twitter.com/tkh44/status/944039519560462336) that you're getting close to shipping Emotion 9! Exciting! Let me know if you have any questions/needs regarding Gatsby!

**Comment by tkh44** [comment]
Let's get this merged and improve it as individual tasks that hopefully the community can help with. 

Besides the terrifying conflicts is there anything I can do to help get this done?

**Pull request merged**